ARS_simulation.py

```diff
@@ -292,9 +294,10 @@ async def simulate_workload_using_predictive_model(function: str, use_await=Fals
             sleep(sleep_time_to_use)
 
     with pr_lock:
-        number_of_parallel_requests_pending -= 1
+        number_of_parallel_requests_pending.value -= 1
 
-    startedCommands.pop(tid)
+    with pr_lock:
+        startedCommands.pop(tid)
 
     with pr_lock:
         for cmd in startedCommands.values():
```

Comment:
Hier drunter, also in Zeile 304 wird der Wert in dem dictionary aktualisiert. Muss anschließend das Objekt nicht neu zugewiesen werden, damit das Dictionary-Objekt mitbekommt, dass eine Änderung stattgefunden hat?
So in der Form:
`startedCommands[i] = cmd` wobei i der index ist?

Siehe hier: https://stackoverflow.com/a/46228938